packages/runner/lib/monitor.ts

```diff
@@ -0,0 +1,156 @@
+        // only report if it has been more than 30 seconds since the last report
+        if (this.lastMemoryReportDate) {
+            const now = new Date();
+            const diffInSecs = (now.getTime() - this.lastMemoryReportDate.getTime()) / 1000;
+            if (diffInSecs < 30) {
+                return;
+            }
+        }
+        this.lastMemoryReportDate = new Date();
+        for (const { environmentId, activityLogId } of this.tracked.values()) {
+            if (!environmentId || !activityLogId) {
+                continue;
+            }
+            await this.send(
+                'post',
+                `/activityLogMessage`,
+                superjson.stringify({
+                    runnerId: this.runnerId,
+                    activityLogMessage: {
+                        level: 'warn',
+                        environment_id: environmentId,
+                        activity_log_id: activityLogId,
+                        content: `Memory usage of nango scripts is high: ${memoryUsagePercentage.toFixed(2)}% of the total available memory.`,
+                        timestamp: Date.now()
+                    } as ActivityLogMessage
+                })
+            );
+        }
+    }
+
+    private checkIdle(): NodeJS.Timeout {
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+        return setInterval(async () => {
+            if (this.tracked.size == 0) {
+                const idleTimeMs = Date.now() - this.lastIdleTrackingDate;
+                if (idleTimeMs > this.idleMaxDurationMs) {
+                    logger.info(`Runner '${this.runnerId}' idle for more than ${this.idleMaxDurationMs}ms`);
+                    await this.send(
+                        'post',
+                        `/idle`,
+                        superjson.stringify({
+                            runnerId: this.runnerId,
+                            idleTimeMs
+                        })
+                    );
+                    this.lastIdleTrackingDate = Date.now();
+                }
+            }
+        }, 10000);
+    }
+
+    private async send(method: string, endpoint: string, superjsonEncodedBody: string): Promise<void> {
+        if (!this.baseUrl) {
+            return;
+        }
+        const url = `${this.baseUrl}${endpoint}`;
+        try {
+            const res = await fetch(url, {
+                method: method,
+                headers: {
+                    Accept: 'application/json',
+                    'Content-Type': 'application/json'
+                },
+                body: superjsonEncodedBody
+            });
+            if (res.status !== 200) {
+                logger.error(`Error reporting '${superjsonEncodedBody}' to '${url}': ${JSON.stringify(await res.json())}`);
+            }
+        } catch (err) {
+            logger.error(`Error reporting '${superjsonEncodedBody}' to '${url}': ${JSON.stringify(err)}`);
```

Comment:
JSON.stringify error doesn't work, I have added an util for that

```suggestion
            logger.error(`Error reporting '${superjsonEncodedBody}' to '${url}': ${stringifyError(err)}`);
```